validate/src/main.rs

```diff
@@ -0,0 +1,51 @@
+use std::fs;
+use std::io::{self, Write};
+use std::path::PathBuf;
+
+use bootspec::generation::Generation;
+use bootspec::Result;
+
+fn main() -> Result<()> {
+    if let Err(e) = self::cli() {
+        writeln!(io::stderr(), "{}", e)?;
+
+        std::process::exit(1);
+    }
+
+    Ok(())
+}
+
+fn cli() -> Result<()> {
+    let mut args = std::env::args().skip(1);
+
+    if args.len() != 1 {
+        writeln!(io::stderr(), "Usage: validate <bootspec_path>")?;
+        std::process::exit(1);
+    }
+
+    let bootspec_path = args
+        .next()
+        .ok_or("Expected path to bootspec document, got none.")?
+        .parse::<PathBuf>()?;
```

Comment:
I think this should be factored out into it's own function to make it easier to read what's going on in this function.

Reply:
https://github.com/DeterminateSystems/bootspec/pull/10 @cole-h